Author: Maximiliano Curia <maxy@debian.org>
Description: Fall back to Artin left_normal_form when libbraiding is not available
Index: sagemath/src/sage/groups/braid.py
===================================================================
--- sagemath.orig/src/sage/groups/braid.py
+++ sagemath/src/sage/groups/braid.py
@@ -74,6 +74,7 @@ from sage.categories.action import Actio
 from sage.categories.groups import Groups
 from sage.combinat.permutation import Permutation, Permutations
 from sage.combinat.subset import Subsets
+from sage.features import FeatureNotPresentError
 from sage.features.sagemath import sage__libs__braiding
 from sage.functions.generalized import sign
 from sage.groups.artin import FiniteTypeArtinGroup, FiniteTypeArtinGroupElement
@@ -1547,9 +1548,12 @@ class Braid(FiniteTypeArtinGroupElement)
             (s0*s1*s0,)
         """
         if algorithm == 'libbraiding':
-            lnf = leftnormalform(self)
-            B = self.parent()
-            return tuple([B.delta()**lnf[0][0]] + [B(b) for b in lnf[1:]])
+            try:
+                lnf = leftnormalform(self)
+                B = self.parent()
+                return tuple([B.delta()**lnf[0][0]] + [B(b) for b in lnf[1:]])
+            except (FeatureNotPresentError, ImportError):
+                return FiniteTypeArtinGroupElement.left_normal_form.f(self)
         if algorithm == 'artin':
             return FiniteTypeArtinGroupElement.left_normal_form.f(self)
         raise ValueError("invalid algorithm")
